scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,311 @@
+from scipy.optimize import minimize_scalar
+from scipy.stats._common import ConfidenceInterval
+from scipy.stats._qmc import check_random_state
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import DecimalNumber, SeedType
+    from typing import Literal
+
+
+__all__ = [
+    'dunnett'
+]
+
+
+@dataclass
+class DunnettResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+    _rho: np.ndarray
+    _df: int
+    _std: float
+    _observations_mean: np.ndarray
+    _control_mean: np.ndarray
+
+    def _allowance(self, confidence_level: DecimalNumber = 0.95) -> float:
+        """Allowance.
+
+        It is the quantity to add/substract from the observed difference
+        between the means of observed groups and the mean of the control
+        group. The result gives confidence limits.
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is .95.
+
+        Returns
+        -------
+        allowance : float
+            Allowance around the mean.
+        """
+        dist = stats.multivariate_t(shape=self._rho, df=self._df)
+        alpha = 1 - confidence_level
+
+        def pvalue_from_stat(statistic):
+            # two-sided to have +- bounds
+            sf = 1 - dist.cdf(statistic, lower_limit=-statistic)
+            return abs(sf - alpha)
+
+        # scipy.stats.sampling methods are not working for this distribution
+        res = minimize_scalar(pvalue_from_stat, method='brent', tol=1e-4)
+        critical_value = res.x
+
+        allowance = critical_value*self._std*np.sqrt(2/len(self.pvalue))
+        return allowance
+
+    def confidence_interval(
+        self, confidence_level: DecimalNumber = 0.95
+    ) -> ConfidenceInterval:
+        """Compute the confidence interval for the specified confidence level.
+
+        The confidence interval corresponds to the mean of the groups
+        +- the allowance.
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is .95.
+
+        Returns
+        -------
+        ci : ``ConfidenceInterval`` object
+            The object has attributes ``low`` and ``high`` that hold the
+            lower and upper bounds of the confidence intervals for each
+            comparison. The high and low values are accessible for each
+            comparison at index ``(i, j)`` between groups ``i`` and ``j``.
+
+        """
+        allowance = self._allowance(confidence_level=confidence_level)
+        diff_means = self._observations_mean - self._control_mean
+        return ConfidenceInterval(
+            low=diff_means-allowance,
+            high=diff_means+allowance
+        )
+
+
+def dunnett(
+    *observations: npt.ArrayLike, control: npt.ArrayLike,
```

Comment:
```suggestion
    *samples: npt.ArrayLike, control: npt.ArrayLike,
```
like other multi-sample tests.